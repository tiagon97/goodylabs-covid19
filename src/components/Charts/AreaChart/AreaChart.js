import React from 'react';
import { AreaChart, Area, XAxis, YAxis, CartesianGrid, Tooltip, ResponsiveContainer, Legend, Text } from 'recharts';
import Heading from '../../Heading/Heading';
const monthNames = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
];

const SimpleAreaChart = ({ singleCountry }) => {
  const formatXAxis = (tickItem) => {
    const month = monthNames[tickItem.slice(6, 7) - 1];
    return `${month} ${tickItem.slice(8, 10)}`;
  };

  return (
    <>
      <Heading>{singleCountry[0].Country}</Heading>
      <ResponsiveContainer width="100%" height={600}>
        <AreaChart data={singleCountry} margin={{ top: 60, right: 80, left: 80, bottom: 60 }}>
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis dataKey="Date" tickFormatter={formatXAxis} />
          <YAxis />
          <Tooltip />
          <Legend verticalAlign="top" height={36} iconType="rect" iconSize={30} />
          <Area type="monotone" dataKey="Confirmed" stroke="#DAA520" fill="#DAA520" />
          <Area type="monotone" dataKey="Recovered" stroke="#32CD32" fill="#32CD32" />
          <Area type="monotone" dataKey="Deaths" stroke="red" fill="red" />
        </AreaChart>
      </ResponsiveContainer>
    </>
  );
};
export default SimpleAreaChart;
