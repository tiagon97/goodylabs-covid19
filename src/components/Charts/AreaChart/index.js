import React from 'react';
import PropTypes from 'prop-types';
import dayjs from 'dayjs';
import {
  AreaChart as RechartsAreaChart,
  Area,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  ResponsiveContainer,
  Legend,
} from 'recharts';

import Heading from '../../Heading';
import formatValue from '../../../utils';
import { Colors } from '../../../constants';

const AreaChart = ({ singleCountry }) => (
  <>
    <Heading>{singleCountry[0].Country}</Heading>
    <ResponsiveContainer width="100%" height={600}>
      <RechartsAreaChart data={singleCountry} margin={{ top: 60, right: 80, left: 80, bottom: 60 }}>
        <CartesianGrid strokeDasharray="3 3" />
        <XAxis
          dataKey="Date"
          tickFormatter={(value) => dayjs(value).format('DD MMM')}
          tick={{ fontSize: 10, transform: 'translate(0, 5)' }}
          minTickGap={-5}
        />
        <YAxis />
        <Tooltip
          formatter={(value) => formatValue`${value}`}
          labelFormatter={(label) => dayjs(label).format('DD MMM')}
        />
        <Legend verticalAlign="top" height={36} iconType="rect" iconSize={30} />
        <Area type="monotone" dataKey="Confirmed" stroke={Colors.yellow} fill={Colors.yellow} />
        <Area type="monotone" dataKey="Recovered" stroke={Colors.green} fill={Colors.green} />
        <Area type="monotone" dataKey="Deaths" stroke={Colors.red} fill={Colors.red} />
      </RechartsAreaChart>
    </ResponsiveContainer>
  </>
);

AreaChart.propTypes = {
  singleCountry: PropTypes.arrayOf(PropTypes.object),
};

AreaChart.defaultProps = {
  singleCountry: [],
};

export default AreaChart;
