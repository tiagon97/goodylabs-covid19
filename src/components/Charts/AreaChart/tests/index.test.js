/**
 *
 * Tests for AreaChart
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import AreaChart from '../index';

const defaultProps = {
  singleCountry: [
    {
      Active: 8454,
      City: '',
      CityCode: '',
      Confirmed: 10892,
      Country: 'Poland',
      CountryCode: '',
      Date: '2020-04-24T00:00:00Z',
      Deaths: 494,
      Lat: '0',
      Lon: '0',
      Province: '',
      Recovered: 1944,
    },
    {
      Active: 8623,
      City: '',
      CityCode: '',
      Confirmed: 11273,
      Country: 'Poland',
      CountryCode: '',
      Date: '2020-04-25T00:00:00Z',
      Deaths: 524,
      Lat: '0',
      Lon: '0',
      Province: '',
      Recovered: 2126,
    },
  ],
};

describe('<AreaChart />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<AreaChart {...defaultProps} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<AreaChart {...defaultProps} />);
    expect(firstChild).toMatchSnapshot();
  });
});
