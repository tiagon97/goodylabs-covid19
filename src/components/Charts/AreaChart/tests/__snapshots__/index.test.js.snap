// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AreaChart /> Should render and match the snapshot 1`] = `
<h1
  class="sc-AxjAm cYlDMx"
>
  Poland
</h1>
`;
